### ISSUE TITLE
Google Chrome will not route to some? sites on the internet when using exit node (Safari & curl work fine)

### ISSUE THREAD

**Issue by billimek** [tailscale/tailscale#7379]
### What is the issue?

Using the exit-node feature of tailscale running on opnsense router.  When my client (tailscale v1.36.1 running on MacOS 13.2.1) uses Google Chrome to reach an internet site (e.g. https://www.google.com/) with exit router enabled, it yields an error: `ERR_CONNECTION_CLOSED`:

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/6393612/221365871-1f1e9421-db19-42c6-aa04-a687585cbd34.png">

... attempting the same thing in Safar, while still using exit node:
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/6393612/221365897-cd62e1b2-a612-482c-913f-4fa83653fb18.png">

This does not seem to be an issue when loading sites inside the local LAN network where tailscale exit router is running.

This doesn't seem to happen with all sites:

Works:
* https://www.reddit.com
* https://www.github.com

Doesn't work:
* https://www.google.com
* https://www.yahoo.com
* https://www.cnn.com

I attempted to remedy by turning MagicDNS on, but that did not yield and difference.  I also attempted to "Allow Local Network Access" in the tailscale client settings.

Maybe related to #2633?

Further info if that helps:

```
root@OPNsense:/usr/local/etc/rc.d # tailscaled --version
1.36.2
  go version: go1.20.1
root@OPNsense:/usr/local/etc/rc.d # tailscale version
1.36.2
  go version: go1.20.1
root@OPNsense:/usr/local/etc/rc.d # tailscale netcheck

Report:
        * UDP: true
        * IPv4: yes, <redacted>:32302
        * IPv6: no, but OS has support
        * MappingVariesByDestIP: false
        * HairPinning: true
        * PortMapping:
        * Nearest DERP: Miami
        * DERP latency:
                - mia: 17.8ms  (Miami)
                - dfw: 19.5ms  (Dallas)
                - ord: 20.8ms  (Chicago)
                - tor: 29.9ms  (Toronto)
                - nyc: 33.4ms  (New York City)
                - den: 44.7ms  (Denver)
                - lax: 49.4ms  (Los Angeles)
                - sfo: 57.2ms  (San Francisco)
                - sea: 61.9ms  (Seattle)
                - sin:         (Singapore)
                - fra:         (Frankfurt)
                - syd:         (Sydney)
                - blr:         (Bangalore)
                - tok:         (Tokyo)
                - lhr:         (London)
                - sao:         (São Paulo)
                - ams:         (Amsterdam)
                - jnb:         (Johannesburg)
                - par:         (Paris)
                - mad:         (Madrid)
                - hkg:         (Hong Kong)
                - waw:         (Warsaw)
                - dbi:         (Dubai)
                - hnl:         (Honolulu)
```

### Steps to reproduce

1. Ensure that exit node is enabled on a tailscale instance
2. Enable exit node on the client (for MacOS: Tailscale Menubar icon -> Exit Node -> Select tailscale server to enable)
3. Lauch Google Chrome
4. Attempt to navigate to a website hosted on internet

### Are there any recent changes that introduced the issue?

Not certain, can't remember if I ever used this feature previously

### OS

Other

### OS version

FreeBSD OPNsense.home 13.1-RELEASE-p6 FreeBSD 13.1-RELEASE-p6 stable/23.1-n250396-d34cd428508 SMP amd64

### Tailscale version

1.36.2

### Other software

Google Chrome Version 110.0.5481.177 (Official Build) (arm64)

### Bug report

BUG-c5f824a225dfa0352ba589fa36982fb6bcce248bcf98632f9c1de8c283879a28-20230225153135Z-d2a37dc8c3562480

**Comment by DentonGentry** [comment]
Around the time of the bugreport, the exit node is reporting TCP RST from several remote destinations:
```
2023-02-25 15:29:50.383378409 +0000 UTC: proxy connection closed with error: readfrom tcp 162.226.x.y:59852->52.96.x.y:443: read tcp 52.96.x.y:443: connection reset by peer
2023-02-25 15:29:50.383428891 +0000 UTC: netstack: forwarder connection to 52.96.x.y:443 closed
2023-02-25 15:29:50.430379211 +0000 UTC: netstack: forwarder connection to 52.96.x.y:443 closed
2023-02-25 15:29:50.430657353 +0000 UTC: netstack: forwarder connection to 52.96.x.y:443 closed
2023-02-25 15:29:50.430743513 +0000 UTC: [RATELIMIT] format("netstack: forwarder connection to %s closed")
2023-02-25 15:30:07.543840339 +0000 UTC: proxy connection closed with error: readfrom tcp 162.226.x.y:60019->52.96.x.y:443: read tcp 52.96.x.y:443: connection reset by peer
2023-02-25 15:30:07.544139746 +0000 UTC: [RATELIMIT] format("netstack: forwarder connection to %s closed") (1 dropped)
2023-02-25 15:30:07.544185363 +0000 UTC: netstack: forwarder connection to 52.96.x.y:443 closed
2023-02-25 15:30:07.560813036 +0000 UTC: proxy connection closed with error: readfrom tcp 162.226.x.y:59884->13.107.x.y:443: read tcp 13.107.x.y:443: connection reset by peer
```

Tailscale itself never sends an RST, these are being received from the WAN. I'd suggest running tshark or tcpdump on the exit node to investigate why.

As I suspect the next question will be why Safari doesn't experience the same issue: it isn't clear why the TCP RST packets are being generated, nor why the browser would make a difference. I note that all of the IP addresses seen are IPv4, it is possible that Safari manages to use IPv6 instead and avoid whatever is generating the RSTs.

> Maybe related to https://github.com/tailscale/tailscale/issues/2633?

That issue results in negotiation down to QUIC/H2, not loss of connectivity. I doubt MTU is related.

**Comment by billimek** [comment]
@DentonGentry I also tried running an exit node on a different device on my network (ubuntu-based host instead of the opnsense router).  There didn't appear to be a change in behavior.

Running tcpdump and trying to interpret the results seemed to suggest that google chrome was only tying ipv6 type connections to certain sites.

Ultimatley, I was able to resolve the issue and have everything work as expected by taking just one action.  I enabled ipv6 on my opnsense router for the wan interface:
<img width="997" alt="image" src="https://user-images.githubusercontent.com/6393612/221420253-bcb38821-3f76-4de4-8959-8587ef3dcb96.png">

... As soon as I did that, everything works as expected.  So, I don't think this may be a tailscale issue, but will defer to you before closing.
